addons/l10n_ma/views/res_company_views.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record id="res_company_form_l10n_ma_view_company" model="ir.ui.view">
+        <field name="name">res.company.form</field>
+        <field name="model">res.company</field>
+        <field name="inherit_id" ref="account.view_company_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//page[@name='general_info']/group/group/field[@name='company_registry']" position="replace">
+                <field name="company_registry" string="ICE" invisible="country_code != 'MA'"/>
+                <field name="company_registry" invisible="country_code == 'MA'"/>
```

Comment:
I don't know if this is the right way of doing it or no.
I have tried `t-if` but it throws error I believe because of that it's not supported here but I don't know if this true or not.
This is the way that came to my mind.